apps/test/unit/lib/kits/maker/util/discountLogicTest.js

```diff
@@ -0,0 +1,28 @@
+import {expect} from '../../../../../util/configuredChai';
+import {isUnit6IntentionEligible, Unit6Intention} from "@cdo/apps/lib/kits/maker/util/discountLogic";
+
+describe('discountLogic.js', () => {
+  describe('isUnit6IntentionEligible', () => {
+    // For winter/spring 2019, discount codes are eligible for teachers that plan to teach
+    // CSD6 during the 18-19 or 19-20 school years.
```

Comment:
Note this logic on the client is only used for display; this check is still [duplicated on the server here](https://github.com/code-dot-org/code-dot-org/blob/b93ce462aafd72f4e54f7aaceabe2a927c306105/dashboard/app/models/circuit_playground_discount_application.rb#L38) where we actually perform the check before distributing a code.  We could go another step and make sure this logic _only_ exists on the server, but I had this working and figured it was a good first step.